src/snowflake/snowpark/session.py

```diff
@@ -1169,11 +1169,15 @@ def get_req_identifiers_list(
 
                 # Add to packages dictionary
                 if name in result_dict:
-                    if version is not None and result_dict[name] != str(package):
-                        raise ValueError(
-                            f"Cannot add dependency package '{name}=={version}' "
-                            f"because {result_dict[name]} is already added."
-                        )
+                    if version is not None:
+                        added_package_has_version = "==" in result_dict[name]
+                        if added_package_has_version and result_dict[name] != str(package):
```

Comment:
Can we add a comment about what an entry of `result_dict` looks like and what `package` looks like? It's not obvious to me why `result_dict[name] != str(package)` is correct 🤔

Reply:
added a comment to explain how things look